api/swagger.yml

```diff
@@ -2734,6 +2746,47 @@ paths:
         default:
           $ref: "#/components/responses/ServerError"
 
+  /repositories/{repository}/branches/{branch}/cherry-pick:
+    parameters:
+      - in: path
+        name: repository
+        required: true
+        schema:
+          type: string
+      - in: path
+        name: branch
+        required: true
+        schema:
+          type: string
+    post:
+      tags:
+        - branches
+      operationId: cherryPick
+      summary: Cherry-Pick the given reference commit into the given branch
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CherryPickCreation"
+      responses:
+        204:
+          description: successful cherry-pick
+        400:
+          $ref: "#/components/responses/ValidationError"
+        401:
+          $ref: "#/components/responses/Unauthorized"
+        404:
+          $ref: "#/components/responses/NotFound"
+        409:
+          description: Conflict Found
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
```

Comment:
Not in this PR, but ideally we will return some detailed error indication of _what_ conflicts we found.  For a merge the caller can probably at least diff, here it will be harder.

Reply:
I agree, but differing this for now. https://github.com/treeverse/lakeFS/issues/5507